lib/chef/knife/ssh.rb

```diff
@@ -219,23 +219,28 @@ def search_nodes
       # @return [Hash<Symbol, Object>]
       def session_options(host, port, user=nil)
         ssh_config = Net::SSH.configuration_for(host)
-        {}.tap do |opts|
+        ssh_config.tap do |opts|
```

Comment:
That is a much bigger change than simply adding the password argument. /ping @chef/client-core